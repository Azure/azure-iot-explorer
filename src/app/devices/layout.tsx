/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { Route, RouteComponentProps, Redirect } from 'react-router-dom';
import DeviceContentContainer from './deviceContent/components/deviceContentContainer';
import DeviceListContainer from './deviceList/components/deviceListContainer';
import AddDeviceContainer from './deviceList/components/addDevice/components/addDeviceContainer';
import SettingsPaneContainer from '../settings/components/settingsPaneContainer';
import HeaderContainer from '../shared/components/headerContainer';
import { ROUTE_PARTS } from '../constants/routes';

export interface LayoutDataProps {
    hubConnectionString: string;
}

export type LayoutProps = RouteComponentProps & LayoutDataProps;

export class DeviceLayout extends React.Component<LayoutProps> {
    constructor(props: LayoutProps) {
        super(props);
    }

    public render(): JSX.Element {
        const url = this.props.match.url;

        return (
            <>
                {this.props.hubConnectionString ?
                    <div className="app">
                        <HeaderContainer />
                        <div className="content">
                        <SettingsPaneContainer />
                        <main role="main">
                            <Route exact={true} path={`${url}`} component={DeviceListContainer}/>
                            <Route exact={true} path={`${url}/${ROUTE_PARTS.ADD}`} component={AddDeviceContainer} />
                            <Route path={`${url}/${ROUTE_PARTS.DETAIL}/`} component={DeviceContentContainer}/>
                        </main>
                        </div>
                    </div> : <Redirect to="/" />
                }
            </>
        );
    }
}
