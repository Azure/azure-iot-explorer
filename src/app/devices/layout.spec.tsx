/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import { DeviceLayout } from './layout';

describe('components/devices/deviceEvents', () => {

    const pathname = `#/devices/detail/digitalTwins/events/?id=device1&interfaceId=urn:contoso:com:EnvironmentalSensor:1`;

    const location: any = { // tslint:disable-line:no-any
        pathname
    };

    const routerProps: any = { // tslint:disable-line:no-any
        history: {
            location
        },
        location,
        match: {}
    };

    const getComponent = (overrides = {}) => {
        const props = {
            hubConnectionString: '',
            ...routerProps,
            ...overrides
        };
        return (<DeviceLayout {...props} />);
    };

    it('matches snapshot', () => {
        const wrapper = shallow(getComponent());
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Redirect').first().prop('to')).toBe('/');
    });

    it('matches snapshot', () => {
        const wrapper = shallow(getComponent({hubConnectionString: 'testString'}));
        expect(wrapper).toMatchSnapshot();
    });
});
