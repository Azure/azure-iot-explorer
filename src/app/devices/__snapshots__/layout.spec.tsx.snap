// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/devices/deviceEvents matches snapshot 1`] = `
<Fragment>
  <Redirect
    push={false}
    to="/"
  />
</Fragment>
`;

exports[`components/devices/deviceEvents matches snapshot 2`] = `
<Fragment>
  <div
    className="app"
  >
    <Connect(Header) />
    <div
      className="content"
    >
      <withRouter(Connect(SettingsPane)) />
      <main
        role="main"
      >
        <Route
          component={[Function]}
          exact={true}
          path="undefined"
        />
        <Route
          component={[Function]}
          exact={true}
          path="undefined/add"
        />
        <Route
          component={[Function]}
          path="undefined/detail/"
        />
      </main>
    </div>
  </div>
</Fragment>
`;
